server/src/main/java/org/opensearch/snapshots/SnapshotsService.java

```diff
@@ -1351,69 +1352,75 @@ public void onFailure(Exception e) {
             @Override
             protected void doRun() {
                 final String localNodeId = clusterService.localNode().getId();
-                repository.getRepositoryData(ActionListener.wrap(repositoryData -> {
-                    try {
-                        final IndexMetadata indexMetadata = repository.getSnapshotIndexMetaData(
-                            repositoryData,
+                if (remoteStoreIndexShallowCopy == false) {
```

Comment:
```suggestion
                if (!remoteStoreIndexShallowCopy) {
```

Reply:
I have seen generally seen this trend of using == false in OpenSearch project and IMO this is probably more readable than !(boolean). Although I don't mind changing it as well.

Reply:
It's definitely a style discussion and I wouldn't hold up approval for it.  The compiler will probably produce identical bytecode in either case.  You may see it a lot from code written by new(-ish) coders who have never been bit by accidentally using `=` instead of `==`.  The simpler code is, the less likely you are to make errors.

One reason I like the shorter form is it somewhat forces verbose and meaningful variable names (so I can read "if not a remote store index shallow copy" in this case).  Compare that to "if remote store index shallow copy is false" which is more "thinking "value of a variable in code" and less natural semantic logic and let you get away with poor variable naming.

It's generally frowned upon in style guides, e.g., [Sonar RSPEC-1125](https://rules.sonarsource.com/java/RSPEC-1125/) and [Checkstyle SimplifyBooleanExpression](https://checkstyle.sourceforge.io/checks/coding/simplifybooleanexpression.html).  Since I use a Sonar linter in my IDE, everytime I see these, they produce little blue squigglies and suggest I fix them. I tend to fix them when I come across them, but like to avoid avoid adding them into new code.

Again, not a big deal, just a common style preference.

Reply:
thanks @dbwiddis, this does put a perspective. And as I think more, do you think if we should any of the static code analyser option widely used in industry. I have previously used find bugs and found it helpful to a large extent. Although this is a tangential discussion from this PR, but want to know your thoughts to start with and then can jump this idea across more maintainers through a GH issue. This way, we can try to fix all the similar issues.

Reply:
@ashking94 It's a mixed bag.  I've considered proposing we add analysis using Sonar to the project, as it's been extremely helpful on my own open source project, and helps me find a lot of issues (potential NPE are particularly valuable).  And having the Sonar linter in my own IDE is helpful when writing/reviewing PRs.  But some of the recommendations are silly and useless, and some rely on recent JVMs but don't work well with older ones.   So it's hard to do a one-size-fits-all solution.

There's also the fact that we inherited a decades-old codebase and doing massive fixes of style preference isn't the best use of our time.  So I generally just focus on "new code" while an analyzer will find thousands of issues in the older code.

Probably worth a "best practices" documentation somewhere but we have enough issues with flakky tests etc. that putting an emphasis on minor style issues is probably not appropriate right now.